lambdaclass/mirra_backend
# [GH-801] Add slow fields as pools in game configuration

> [!Warning]
> Merge alongside [This client branch](https://github.com/lambdaclass/champions_of_mirra/pull/2011)

## Motivation

We need to implement two new kind of enviromental hazards:
- The map needs to spawn pools that apply an slowness effects to players that go over them
- We need to add a new kind of obstacle that allow projectiles to go through but not players

Closes #801

## Summary of changes

- Added "lake" type of obstacles that allow projectiles to go over them but not players
- Added pools to the map config that will spawn without a defined duration applying a slowness effect to the players
- Added new forms to the game config for easy edit of pools and obstacles in the map configurations

## How to test it?

- Run the seeds  with make start
- Try to play a match in the game client and you should see that two previous obstacles are now pools that slows you down

<img width="1471" alt="Captura de pantalla 2024-07-26 a la(s) 12 30 38 p  m" src="https://github.com/user-attachments/assets/de684e65-75b8-447b-8d4a-110bbd7d8070">

### pools test
Add this to the seeds `pools` array: 

```
pools: [
    %{
      name: "oil pool north a",
      position: %{
        x: 0.0,
        y: 0.0
      },
      radius: 0.0,
      shape: "polygon",
      vertices: [
        %{
          x: 2957.0,
          y: 224.0
        },
        %{
          x: 3160.0,
          y: 295.0
        },
        %{
          x: 3224.0,
          y: 800.0
        },
        %{
          x: 2865.0,
          y: 1222.0
        },
        %{
          x: 2663.0,
          y: 1228.0
        },
        %{
          x: 2527.0,
          y: 1057.0
        },
        %{
          x: 2521.0,
          y: 833.0
        }
      ],
      effects_to_apply: [
        "slow_field"
      ]
    }
  ]
```
Then start a match and there should be some pools in the map that slows you down

### Lake test
Change the type from an obstacle from `static` to `lake` and the projectiles should be able to go through it

## Checklist
